packages/protocol/contracts/governance/DoubleSigningSlasher.sol

```diff
@@ -0,0 +1,179 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./SlasherUtil.sol";
+
+contract DoubleSigningSlasher is SlasherUtil {
+  using SafeMath for uint256;
+
+  struct SlashingIncentives {
+    // Value of LockedGold to slash from the account.
+    uint256 penalty;
+    // Value of LockedGold to send to the observer.
+    uint256 reward;
+  }
+
+  SlashingIncentives public slashingIncentives;
+
+  // For each signer address, check if a block header has already been slashed
+  mapping(address => mapping(bytes32 => bool)) isSlashed;
+
+  event SlashingIncentivesSet(uint256 penalty, uint256 reward);
+
+  /**
+   * @notice Initializer
+   * @param registryAddress Sets the registry address. Useful for testing.
+   * @param _penalty Penalty for the slashed signer.
+   * @param _reward Reward that the observer gets.
+   */
+  function initialize(address registryAddress, uint256 _penalty, uint256 _reward)
+    external
+    initializer
+  {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    setSlashingIncentives(_penalty, _reward);
+  }
+
+  /**
+   * @notice Sets slashing incentives.
+   * @param penalty Penalty for the slashed signer.
+   * @param reward Reward that the observer gets.
+   */
+  function setSlashingIncentives(uint256 penalty, uint256 reward) public onlyOwner {
+    require(penalty > reward, "Penalty has to be larger than reward");
+    slashingIncentives.penalty = penalty;
+    slashingIncentives.reward = reward;
+    emit SlashingIncentivesSet(penalty, reward);
+  }
+
+  /**
+   * @notice Counts the number of set bits (Hamming weight).
+   * @param v Bitmap.
+   * @return Number of set bits.
+   */
+  function countSetBits(uint256 v) internal pure returns (uint256) {
+    uint256 res = 0;
+    uint256 acc = v;
+    for (uint256 i = 0; i < 256; i++) {
+      if (acc & 1 == 1) res++;
+      acc = acc >> 1;
+    }
+    return res;
+  }
+
+  /**
+   * @notice Returns the minimum number of required signers for a given block number.
+   */
+  function minQuorumSize(uint256 blockNumber) internal view returns (uint256) {
+    return (2 * numberValidatorsInSet(blockNumber) + 2) / 3;
+  }
+
+  /**
+   * @notice Given two RLP encoded blocks, calls into precompiles to require that
+   * the two block hashes are different, have the same height, have a
+   * quorum of signatures, and that `signer` was part of the quorum.
+   * @param signer The signer to be slashed.
+   * @param index Validator index at the block.
+   * @param blockA First double signed block.
+   * @param blockB Second double signed block.
+   * @return Block number where double signing occured. Throws if no double signing is detected.
+   */
+  function checkForDoubleSigning(
```

Comment:
Not sure if relevant.... but given 2 blocks on which I, the signer, had sign. Could they be from 2 different networks? like alfajores & baklave? shouldn't we check that?

Reply:
As @jfoutts-celo mentioned on Slack, this is already checked on `getVerifiedSealBitmapFromHeader`. Since it verifies that both block have the same parent hash and that the hash belongs to the current blockchain